r/PACKAGING.md

```diff
@@ -0,0 +1,116 @@
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Packaging checklist for CRAN release
+
+For full documentation of the steps in this checklist, see the
+[Apache Arrow Release Management Guide](https://cwiki.apache.org/confluence/display/ARROW/Release+Management+Guide#ReleaseManagementGuide-UpdatingRpackages).
+
+- [ ] When a release candidate is created
+  [Create a GitHub issue](https://github.com/apache/arrow/issues/new/)
+  entitled `[R] CRAN release version X.X.X` and copy this checklist to the issue
+  
+- [ ] Confirm that all
+  [nightly tests and nightly packaging builds](https://lists.apache.org/list.html?builds@arrow.apache.org) are passing.
+- [ ] Check [current CRAN check results](https://cran.rstudio.org/web/checks/check_results_arrow.html)
+- [ ] Check README
+- [ ] `urlchecker::url_check()`
+- [ ] [Polish NEWS](https://style.tidyverse.org/news.html#news-release)
+- [ ] Prepare tweet thread highlighting new features
+
+Make draft pull requests into the [autobrew](https://github.com/autobrew) and
+[rtools-packages](https://github.com/r-windows/rtools-packages) repositories
+used by the configure script on MacOS and Windows:
+
+- [ ] Open a draft pull request to modify 
+  [the apache-arrow autobrew formula]( https://github.com/autobrew/homebrew-core/blob/master/Formula/apache-arrow.rb) 
+  to update the version, SHA, and any changes to dependencies and build steps that have changed in the
+  [copy of the formula we have of that formula in the Arrow repo](https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow.rb)
+- [ ] Open a draft pull request to modify
+  [the apache-arrow-static autobrew formula]( https://github.com/autobrew/homebrew-core/blob/master/Formula/apache-arrow-static.rb)
+  to update the version, SHA, and any changes to dependencies and build steps that have changed in the
+  [copy of the formula we have of that formula in the Arrow repo](https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow-static.rb)
+- [ ] Open a draft pull request to modify the 
+  [autobrew script](https://github.com/autobrew/scripts/blob/master/apache-arrow)
+  to include any additions made to
+  [r/tools/autobrew](https://github.com/apache/arrow/blob/master/r/tools/autobrew).
+- [ ] Open a draft pull request to modify the
+  [PKGBUILD script](https://github.com/r-windows/rtools-packages/blob/master/mingw-w64-arrow/PKGBUILD)
+  to reflect changes in
+  [ci/PKGBUILD](https://github.com/apache/arrow/blob/master/ci/scripts/PKGBUILD),
+  uncommenting the line that says "uncomment to test the rc".
+
+Prepare and check the .tar.gz that will be released to CRAN.
+
+- [ ] `git fetch upstream && git checkout release-X.X.X-rcXX`
+- [ ] `make build` (copies Arrow C++ into tools/cpp, prunes some unnecessary
+  components, and runs `R CMD build`)
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally on Linux
```

Comment:
Are these necessary? This isn't something we've done before. The release-verification crossbow job could do these. But we generally don't care about source builds on win/mac, and Linux is covered in regular CI.